rust/kernel/error.rs

```diff
@@ -21,44 +21,49 @@ use core::str::{self, Utf8Error};
 ///
 /// The value is a valid `errno` (i.e. `>= -MAX_ERRNO && < 0`).
 #[derive(Clone, Copy, PartialEq, Eq)]
-pub struct Error(c_types::c_int);
+pub struct Error(NonZeroI32);
```

Comment:
If you guys believe that `NonZeroI32` is optimal, perhaps put a comment that explains why?
Otherwise we might get PRs from newcomers who think that `NonZeroI32` occupies too much space, so they'll replace with `NonZeroI16`...